src/io.jl

```diff
@@ -223,9 +226,530 @@ function readg(gfile; set_time=nothing)
     z = range(zmid - 0.5*zdim, zmid + 0.5*zdim, length=nh)
     psi = range(simag, sibry, length=nw)
 
-    g = GEQDSKFile(gfile,time,nw,nh,r,z,rdim,zdim,rleft,zmid,nbbbs,rbbbs,zbbbs,limitr,rlim,zlim,
-                   rcentr,bcentr,rmaxis,zmaxis,simag,sibry,psi,current,fpol,pres,ffprim,pprime,
-                   qpsi,psirz,rhovn)
+    g = GEQDSKFile(
+        splitpath(gfile)[end],time,nw,nh,r,z,rdim,zdim,rleft,zmid,nbbbs,rbbbs,zbbbs,
+        limitr,rlim,zlim,rcentr,bcentr,rmaxis,zmaxis,simag,sibry,psi,current,fpol,pres,
+        ffprim,pprime,qpsi,psirz,rhovn
+    )
 
     return g
 end
+
+geqdsk2imas_generic = """Variations of this function accept single GEQDSKFile instances or vectors of these.
+Also, a complete data dictionary can be provided, or a reference to the equilibrium IDS,
+(both of these needing instruction on which time slice to write to) or to a single
+equilibrium.time\\_slice. If the top level DD isn't referenced, a reference to the
+wall IDS can be passed in separately to allow writing wall data.
+The COCOS of the incoming geqdsk files can be spefificed, or it can be automatically
+detected by setting geqdsk\\_cocos=0 (the default). If multiple COCOS are possible,
+the first one in the list will be chosen, which usually means 3 intead of 13, for example.
+To narrow down the choice of COCOS, it is necessary to specify the direction of toroidal
+angle using cocos\\_clockwise\\_phi, which defaults to false (meaning phi is CCW).
```

Comment:
@eldond gEQDSKs can be in either COCOS 1,3,5,7 depending on the direction of the current and magnetic field. Please take a look at https://github.com/gafusion/OMFIT-source/blob/92a14adeca963b6fe58520cd61f536be1ddee587/omfit/omfit_classes/omfit_eqdsk.py#L3263-L3293

I feel the `geqdsk2imas_generic` text does not add much to what's already in the docstrings without that text... no? Also, be conscious that the docstrings should look nice in MarkDown format, so maybe something like this format would look nicer when rendered in the online documentation:
```
    geqdsk2imas!(
        gs::Vector{GEQDSKFile},
        dd::IMASdd.dd;
        geqdsk_cocos::Int=0,
        dd_cocos::Int=11,
        cocos_clockwise_phi::Bool=false,
        add_derived::Bool=false)

Utility for writing equilibrium data from GEQDSKFiles to IMAS schema.

* gs: Vector of GEQDSKFile instances
* dd: Top level IMAS data dictionary
* geqdsk_cocos: coordinate convention indentifier of the incoming GEQDSK set. 0 to auto detect.
* dd_cocos: coordinate convention identifier of the DD. Should normally be left at 11.
* cocos_clockwise_phi: hint for deciding COCOS, because GEQDSK probably doesn't constrain this.
* add_derived: switch to use functions within EFIT.jl to extend equilibrium data
```

Notice that the docstring should not have `function geqdsk2imas!`and just `geqdsk2imas!` suffices